projects/plugins/boost/compatibility/wp-super-cache.php

```diff
@@ -40,17 +43,27 @@ function clear_cache() {
 	}
 }
 add_action( 'jetpack_boost_critical_css_generated', __NAMESPACE__ . '\clear_cache' );
+add_action( 'jetpack_boost_critical_css_invalidated', __NAMESPACE__ . '\clear_cache' );
 
 /**
  * Clear Super Cache's cache when a module is enabled or disabled.
  *
- * @param string $module The module slug.
+ * @param string $module_slug The module slug.
  * @param bool   $status The new status.
  */
-function module_status_updated( $module, $status ) {
+function module_status_updated( $module_slug, $status ) {
+	// Get a list of modules that can change the HTML output.
+	$output_changing_modules = Modules_Index::get_modules_implementing( Changes_Output::class );
+
 	// Special case: don't clear when enabling Critical or Cloud CSS, as they will
 	// be handled after generation.
-	if ( $status && ( $module === 'critical-css' || $module === 'cloud-css' ) ) {
+	if ( $status ) {
+		unset( $output_changing_modules['critical_css'] );
```

Comment:
This makes me really want to consider `Affects_Speed_Score` as an interface name - it's very specific and unmistakeable for what it actually implies.